Python extension for PGPLOT-style (scientific) data plotting

haavee opened the pull request:
<!--
Thank you very much for putting in this recipe PR!

This repository is very active, so if you need help with a PR, please let the
right people know. There are language-specific teams for reviewing recipes.

| Language        | Name of review team           |
| --------------- | ----------------------------- |
| python          | `@conda-forge/help-python`    |
| python/c hybrid | `@conda-forge/help-python-c`  |
| r               | `@conda-forge/help-r`         |
| java            | `@conda-forge/help-java`      |
| nodejs          | `@conda-forge/help-nodejs`    |
| c/c++           | `@conda-forge/help-c-cpp`     |
| perl            | `@conda-forge/help-perl`      |
| Julia           | `@conda-forge/help-julia`     |
| ruby            | `@conda-forge/help-ruby`      |
| Rust            | `@conda-forge/help-rust`      |
| Go              | `@conda-forge/help-go`        |
| other           | `@conda-forge/staged-recipes` |

Once the PR is ready for review, please mention one of the teams above in a
new comment. i.e. `@conda-forge/help-some-language, ready for review!`
Then, a bot will label the PR as 'review-requested'.

Due to GitHub limitations, first time contributors to conda-forge are unable
to ping conda-forge teams directly, but you can [ask a bot to ping the team][1]
using a special command in a comment on the PR to get the attention of the
`staged-recipes` team. You can also consider asking on our [Zulip chat][2]
if your recipe isn't reviewed promptly.

[1]: https://conda-forge.org/docs/maintainer/infrastructure.html#conda-forge-admin-please-ping-team
[2]: https://conda-forge.zulipchat.com

All apologies in advance if your recipe PR does not receive prompt attention.
This is a high volume repository and the reviewers are volunteers. Review times
vary depending on the number of reviewers on a given language team and may be days
or weeks. We are always looking for more staged-recipe reviewers. If you are
interested in volunteering, please contact a member of @conda-forge/core.
We'd love to have the help!
-->

Checklist
- [x] Title of this PR is meaningful: e.g. "Adding my_nifty_package", not "updated meta.yaml".
- [x] License file is packaged (see [here](https://github.com/conda-forge/staged-recipes/blob/5eddbd7fc9d1502169089da06c3688d9759be978/recipes/example/meta.yaml#L64-L73) for an example).
- [x] Source is from official source.
- [x] Package does not vendor other packages. (If a package uses the source of another package, they should be separate packages or the licenses of all packages need to be packaged).
- [x] If static libraries are linked in, the license of the static library is packaged.
- [x] Package does not ship static libraries. If static libraries are needed, [follow CFEP-18](https://github.com/conda-forge/cfep/blob/main/cfep-18.md).
- [x] Build number is 0.
- [x] A tarball (`url`) rather than a repo (e.g. `git_url`) is used in your recipe (see [here](https://conda-forge.org/docs/maintainer/adding_pkgs.html#build-from-tarballs-not-repos) for more details).
- [x] GitHub users listed in the maintainer section have posted a comment confirming they are willing to be listed there.
- [x] When in trouble, please check our [knowledge base documentation](https://conda-forge.org/docs/maintainer/knowledge_base.html) before pinging a team.

Commit:
Add recipe for python-pgplot

conda-forge-admin commented:
Hi! This is the friendly automated conda-forge-linting service.

I wanted to let you know that I linted all conda-recipes in your PR (```recipes/python-pgplot/meta.yaml```) and found some lint.

Here's what I've got...


For **recipes/python-pgplot/meta.yaml**:

 * ❌ Selectors are suggested to take a ``<two spaces>#<one space>[<expression>]`` form. See lines [19]
 * ❌ This recipe is using a compiler, which now requires adding a build dependence on `{{ stdlib("c") }}` as well. Note that this rule applies to each output of the recipe using a compiler. For further details, please see https://github.com/conda-forge/conda-forge.github.io/issues/2102.

For **recipes/python-pgplot/meta.yaml**:

 * ℹ️ PyPI default URL is now pypi.org, and not pypi.io. You may want to update the default source url.


<sub>This message was generated by GitHub Actions workflow run [https://github.com/conda-forge/conda-forge-webservices/actions/runs/16930953803](https://github.com/conda-forge/conda-forge-webservices/actions/runs/16930953803). Examine the logs at this URL for more detail.</sub>

Commit:
Fix conda-auto-linter warnings-cum-errors

conda-forge-admin commented:
Hi! This is the friendly automated conda-forge-linting service.

I just wanted to let you know that I linted all conda-recipes in your PR (```recipes/python-pgplot/meta.yaml```) and found it was in an excellent condition.

Commit:
Drop libc 2.12, x11-* dep only goes down to 2.17

Commit:
Resolve over-/underspecifying runtime dependencies

conda-forge-admin commented:
Hi! This is the friendly automated conda-forge-linting service.

I wanted to let you know that I linted all conda-recipes in your PR (```recipes/python-pgplot/meta.yaml```) and found some lint.

Here's what I've got...


For **recipes/python-pgplot/meta.yaml**:

 * ❌ If python is a host requirement, it should be a run requirement.


<sub>This message was generated by GitHub Actions workflow run [https://github.com/conda-forge/conda-forge-webservices/actions/runs/16931841225](https://github.com/conda-forge/conda-forge-webservices/actions/runs/16931841225). Examine the logs at this URL for more detail.</sub>

Commit:
Conda-lint: Python host req? The also run req

conda-forge-admin commented:
Hi! This is the friendly automated conda-forge-linting service.

I just wanted to let you know that I linted all conda-recipes in your PR (```recipes/python-pgplot/meta.yaml```) and found it was in an excellent condition.

Commit:
The lib is called freetype6, not freetype

Commit:
Maybe it's even called libfreetype6 ...

According to
https://anaconda.org/conda-forge/libfreetype6

Commit:
Builds fail on missing LICENSE file (fixed)

haavee commented:
Hi all, sorry for the splurge of small commits - I don't have access to a test platform that can run `build-locally.py`
Before generating the PR I _did_ verify that `conda build purge && conda build conda-recipe/ -c conda forge` on my ARM64 MacOS and Linux AMD64  both built without errors.

Whilst I'm at it, I hereby ACK the following:
> GitHub users listed in the maintainer section have posted a comment confirming they are willing to be listed there

haavee commented:
@ conda-forge/help-python-c I think the recipe is ready for review

haavee commented:
@conda-forge-admin, please ping team

traversaro commented on recipes/python-pgplot/bld.bat:
```diff
@@ -0,0 +1,5 @@
+REM Windows build script - currently not supported due to giza dependency
+echo "Windows builds are not currently supported for python-pgplot"
+echo "This is due to the giza dependency not being available on Windows"
+exit /b 1
```
If Windows is skipped in the meta.yaml, you can just do not add a `bld.bat`.

traversaro commented on recipes/python-pgplot/conda_build_config.yaml:
```diff
@@ -0,0 +1,9 @@
+  - macosx_deployment_target   # [osx]
+c_stdlib_version:              # [unix]
+  - 2.17                       # [linux and x86_64]
+  - 11.0                       # [osx and arm64]
```
Why you need a custom one, can't you just use the official conda-forge-pinning ?

traversaro commented on recipes/python-pgplot/build.sh:
```diff
@@ -0,0 +1,39 @@
+export CFLAGS="-fPIC"
+export CXXFLAGS="-fPIC"
+export LDFLAGS="-L${PREFIX}/lib"
+export CPPFLAGS="-I${PREFIX}/include"
```
I am missing something, why are we building giza from source if it is available from conda-forge?

carterbox commented:
Waiting for author to request an apple silicon build of a dependency "giza" instead of revending it here.

haavee commented:
Thanks for your effort!

I have put in a request to the "giza" team for an osx-arm64 build, but most importantly asked if the install includes the "include files" of giza.  Without the, let's say "giza-dev" package (which under Linux is the name of the package that installs headers, right?) this Python binding won't compile.

Let's see what happens next.

haavee committed:
Merge branch 'main' into python-pgplot

Commit:
Update recipe to rely on conda-forge giza library

Thanks to the giza feedstock maintainer's effort to add & fix OSX-ARM64
build of giza the recipe now only relies on other conda-forge libraries
and should compile & run on both Linux and OSX

conda-forge-admin commented:
Hi! This is the friendly automated conda-forge-linting service.

I wanted to let you know that I linted all conda-recipes in your PR (```recipes/python-pgplot/meta.yaml```) and found some lint.

Here's what I've got...


For **recipes/python-pgplot/meta.yaml**:

 * ❌ Selectors are suggested to take a ``<two spaces>#<one space>[<expression>]`` form. See lines [19]
 * ❌ This recipe is using a compiler, which now requires adding a build dependence on `{{ stdlib("c") }}` as well. Note that this rule applies to each output of the recipe using a compiler.


<sub>This message was generated by GitHub Actions workflow run [https://github.com/conda-forge/conda-forge-webservices/actions/runs/26021862174](https://github.com/conda-forge/conda-forge-webservices/actions/runs/26021862174). Examine the logs at this URL for more detail.</sub>

Commit:
Address linter remarks / suggestinos

conda-forge-admin commented:
Hi! This is the friendly automated conda-forge-linting service.

I just wanted to let you know that I linted all conda-recipes in your PR (```recipes/python-pgplot/meta.yaml```) and found it was in an excellent condition.

haavee commented:
I think that the recipe now passes muster:
- only depends on conda-forge libraries
- compiles & tests on Linux and OSX cleanly
- the recipe is accepted by the linter

So, what next?
